File: media/textsamples/sample.swift
// sample.swift — Swift syntax-highlighting sample for the UltraCanvas demo.
import Foundation

protocol Shape {
    var name: String { get }
    func area() -> Double
}

struct Circle: Shape {
    let radius: Double

    var name: String { "circle" }

    func area() -> Double {
        Double.pi * radius * radius
    }
}

func totalArea(_ shapes: [Shape]) -> Double {
    shapes.reduce(0) { $0 + $1.area() }
}

let shapes: [Shape] = [Circle(radius: 1.0), Circle(radius: 2.5)]

for shape in shapes {
    print(String(format: "%@ -> %.4f", shape.name, shape.area()))
}

print("total = \(totalArea(shapes))")

File: media/textsamples/sample.ts
// sample.ts — TypeScript syntax-highlighting sample for the UltraCanvas demo.

interface Shape {
    readonly name: string;
    area(): number;
}

type Color = "red" | "green" | "blue";

class Circle implements Shape {
    constructor(public readonly radius: number, private color: Color = "blue") {}

    get name(): string {
        return "circle";
    }

    area(): number {
        return Math.PI * this.radius ** 2;
    }
}

function totalArea(shapes: readonly Shape[]): number {
    return shapes.reduce((sum, s) => sum + s.area(), 0);
}

const shapes: Shape[] = [new Circle(1), new Circle(2.5, "red")];
console.log(`total area = ${totalArea(shapes).toFixed(3)}`);

const enum Direction { Up, Down, Left, Right }
const move = (d: Direction): string => Direction[d];
console.log(move(Direction.Left));
